### PR TITLE
Adding Push To Try documentation

### PR DESCRIPTION
**Pull request by daveccrit**
r? @Standard8, @dmose 

Likely will need some changes, but this is what I have been doing for push to try.

### CODE REVIEW

**Comment by mozilla-autolander-deprecated** [comment]
Autolander could not find a bug number in your pull request title. All pull requests should be in the format of: Bug [number] - [description].

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -93,6 +93,12 @@ When practical:
 - The images are optimized
 - It should be (keyboard- and/or ARIA-) accessible
 
+Testing new Mochitests:
+While it is usually sufficient to test new mochitests locally, some tests 
+may fail running on other platforms. Pushing your new tests to the Try 
+server is necessary before landing. See [Push-To-Try.md](docs/Push-To-Try.md)
```
Comment (Standard8) [inline CONTRIBUTING.md:99]: Can you also update the section in Developing.md please?

Comment (Standard8) [inline CONTRIBUTING.md:99]: Maybe to reference the `make git-export` (with GIT_EXPORT_LOCATION) and then also link to the push-to-try doc.

File: docs/Push-To-Try.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+============
+
+Commit your local changes
+On Local Loop Branch:
+```shell
+$ git commit -m “Bug 123456789 - whatever message you want” .
+```
+
+Update fx-team on gecko-dev to latest, remove old branch if exists and create new branch for your changes
+On gecko-dev/fx-team:
```
Comment (Standard8) [inline docs/Push-To-Try.md:35]: Markdown is putting these two lines on the same line - I think you need an additional blank line between them (in several places).

File: docs/Push-To-Try.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+```
+
+Create commit in testBranch on gecko-dev. Determine which commands to use for try server push and then execute try server push.
+
+* Use http://trychooser.pub.build.mozilla.org/ to get command line for try server push
+
+On gecko-dev/testBranch:
+```shell
+$ git commit -m “Test changes for Try server” .
+$ git-push-to-try -t -r HEAD ../firefox "-b do -p linux64 -u mochitest-bc,mochitest-e10s-bc -t none"
```
Comment (Standard8) [inline docs/Push-To-Try.md]: You're using `mozilla-central` as the example, so you need to change the `../firefox` here.

File: docs/Push-To-Try.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+On Local Loop Branch:
+```shell
+$ make build
+$ make git-export
+```
+
+Update mozilla-central to latest
+On mozilla-central:
+```shell
+$ hg pull
```
Comment (Standard8) [inline docs/Push-To-Try.md]: I'd suggest adding a `cd ../mozilla-central` just before this, generally makes it clearer.

**Comment by mozilla-autolander-deprecated** [comment]
Autolander could not find a bug number in your pull request title. All pull requests should be in the format of: Bug [number] - [description].

**Comment by mozilla-autolander-deprecated** [comment]
Autolander could not find a bug number in your pull request title. All pull requests should be in the format of: Bug [number] - [description].

**Comment by Standard8** [comment]
@daveccrit I've rebased the branch onto master and added another commit to address my review comments. If you can take a look through and check they are OK, then you can have r=Standard8 to squash and merge these into master.

**Comment by mozilla-autolander-deprecated** [comment]
Autolander could not find a bug number in your pull request title. All pull requests should be in the format of: Bug [number] - [description].

**Pull request merged**